src/btree/bt_split.c

```diff
@@ -1433,8 +1433,15 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
     __wt_btcur_init(session, &cbt);
     __wt_btcur_open(&cbt);
 
+    /* If we have saved updates, we must have decided to restore them to the new page. */
+    WT_ASSERT(session, multi->supd_entries == 0 || multi->supd_restore);
+
```

Comment:
This doesn't look right to me, at line 1427 we checked `if (multi->supd_entries == 0)` and returned.